### ISSUE TITLE
Compile error on rustc 1.23.0-nightly (d6b06c63a 2017-11-09)

### ISSUE THREAD

**Issue by topecongiro** [rust-lang-nursery/rust-clippy#2221]
`cargo install clippy` fails with `rustc 1.23.0-nightly (d6b06c63a 2017-11-09)`.
```
error[E0560]: struct `rustc_errors::CodeSuggestion` has no field named `substitution_parts`
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.169/src/utils/mod.rs:615:9
    |
615 |         substitution_parts: sugg.into_iter()
    |         ^^^^^^^^^^^^^^^^^^^ field does not exist - did you mean `substitutions`?

error[E0560]: struct `rustc_errors::Substitution` has no field named `span`
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.169/src/utils/mod.rs:618:21
    |
618 |                     span: span,
    |                     ^^^^^ `rustc_errors::Substitution` does not have this field
    |
    = note: available fields are: `parts`

error[E0560]: struct `rustc_errors::Substitution` has no field named `substitutions`
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.169/src/utils/mod.rs:619:21
    |
619 |                     substitutions: vec![sub],
    |                     ^^^^^^^^^^^^^^ `rustc_errors::Substitution` does not have this field
    |
    = note: available fields are: `parts`

error[E0608]: cannot index into a value of type `std::rc::Rc<std::collections::HashSet<rustc::hir::ItemLocalId, std::hash::BuildHasherDefault<rustc_data_structures::fx::FxHasher>>>`
   --> .cargo/registry/src/github.com-1ecc6299db9ec823/clippy_lints-0.0.169/src/methods.rs:890:26
    |
890 |           let promotable = cx.tcx
    |  __________________________^
891 | |             .rvalue_promotable_map(owner_def)
892 | |             [&arg.hir_id.local_id];
    | |__________________________________^

error: aborting due to 4 previous errors

error: Could not compile `clippy_lints`.
```

**Comment by oli-obk** [comment]
That's all my fault, even though from different PRs ;) I'm on it.

**Comment by oli-obk** [comment]
Fixed in 299f1270a6d0dfa9b77ab0c61b1f25f257a9b692
